hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreScanner.java

```diff
@@ -786,6 +791,21 @@ private void updateMetricsStore(boolean memstoreRead) {
     }
   }
 
+  private void doSeekCol(Cell cell) throws IOException {
+    // we check when ever a seek_next_col happens did the seek really land in a new block.
+    // If the seek always lands in the same current block while trying to do a next(),
+    // we tend to go with next() rather than seek() based on the 'seekToSameBlock'
+    // which is updated in the method 'seekOrSkipToNextColumn'. Do this when rowColBloom
+    // is not used
+    if (this.matcher.isUserScan() && !get && !useRowColBloom && seekToSameBlock
+        && bytesRead > switchToNextOnlyBytes) {
```

Comment:
Why need "bytesRead > switchToNextOnlyBytes"? Can do it directly?

Reply:
@infraio  - Thanks for the review. Can you tell me more when you say 'do it directly'? Without observing for few blocks how to know whether the addColumns is actually projecting all/most of the columns or say it is projecting in such a way that instead of the nextblock we land in next+1 block()? So this is more like an observation we do and then decide if to take an action based on that observation.

Reply:
this.switchToNextOnlyBytes = this.preadMaxBytes. Then why use preadMaxBytes to decide this?

Reply:
Oh I got your point.  The reason to do that was to see if we can introduce a new config for that instead of using preadMaxBytes config. I have not added a config (ie. a TODO). That config can be for now directly referring to preadMaxBytes or may be make it a new value.